src/GitLink/Linker.cs

```diff
@@ -23,234 +26,81 @@ namespace GitLink
+                    provider = candidateProviders.First();
                 }
-                catch (Exception ex)
+                else
                 {
-                    Log.Error(ex, "An unexpected error occurred");
+                    provider = providerManager.GetProvider(options.GitRemoteUrl.AbsoluteUri);
                 }
 
-                stopWatch.Stop();
-            }
-
-            Log.Info(string.Empty);
-            Log.Info("Completed in '{0}'", stopWatch.Elapsed);
+                var projectSrcSrvFile = pdbPath + ".srcsrv";
 
-            exitCode = exitCode ?? -1;
-
-            if (context.ErrorsAsWarnings && exitCode != 0)
-            {
-                Log.Info("One or more errors occurred, but treating it as warning instead");
-
-                exitCode = 0;
-            }
-
-            return exitCode.Value;
-        }
-
-        private static bool LinkProject(Context context, Project project, string pdbStrFile, string shaHash, string pathPdbDirectory = null)
-        {
-            Argument.IsNotNull(() => context);
-            Argument.IsNotNull(() => project);
-
-            try
-            {
-                var projectName = project.GetProjectName();
-
-                Log.Info("Handling project '{0}'", projectName);
-
-                Log.Indent();
-
-                var compilables = project.GetCompilableItems().Select(x => x.GetFullFileName()).ToList();
-
-                var outputPdbFile = project.GetOutputPdbFile();
-                var projectPdbFile = pathPdbDirectory != null ? Path.Combine(pathPdbDirectory, Path.GetFileName(outputPdbFile)) : Path.GetFullPath(outputPdbFile);
-                var projectSrcSrvFile = projectPdbFile + ".srcsrv";
-
-                var srcSrvContext = new SrcSrvContext
-                {
-                    Revision = shaHash,
-                    RawUrl = context.Provider.RawGitUrl,
-                    DownloadWithPowershell = context.DownloadWithPowershell
-                };
-
-                if (!File.Exists(projectPdbFile))
-                {
-                    Log.Warning("No pdb file found for '{0}', is project built in '{1}' mode with pdb files enabled? Expected file is '{2}'", projectName, context.ConfigurationName, projectPdbFile);
-                    return false;
-                }
-
-                if (!context.SkipVerify)
+                if (!options.SkipVerify)
                 {
                     Log.Info("Verifying pdb file");
 
-                    var missingFiles = project.VerifyPdbFiles(compilables, projectPdbFile);
+                    var missingFiles = pdb.FindMissingOrChangedSourceFiles();
                     foreach (var missingFile in missingFiles)
                     {
-                        Log.Warning("Missing file '{0}' or checksum '{1}' did not match", missingFile.Key, missingFile.Value);
+                        Log.Warning($"File \"{missingFile}\" missing or changed since the PDB was compiled.");
```

Comment:
Use single quotes for the values in logging please (to be consistent with the rest of the app).